### ISSUE TITLE
File upload integration

### ISSUE THREAD

**Issue by martinos** [boydm/phoenix_integration#2]
I am currently trying to test a form that has a file upload field. Is there a way to test these form using the `follow_form` function ? 

I tried the following code: 

``` elixir
    upload = %Plug.Upload{content_type: "text", path: "mix.exs", filename: "mix.exs"}

    get(conn, expense_path(conn, :index) )
    |> follow_form(%{ expense: %{receipt: upload, description: "Cell"}})
    |> assert_response(status: 200, path: expense_path(conn, :index), html: "Cell")
```

And I got the following message:

```
  1) test Expense page flow (CieExpenses.ExpenseIntegrationTest)
     test/integrations/expense_integration_test.exs:4
     ** (Protocol.UndefinedError) protocol Enumerable not implemented for %Plug.Upload{content_type: "text", filename: "mix.exs", path: "mix.exs"}
     stacktrace:
       (elixir) lib/enum.ex:1: Enumerable.impl_for!/1
       (elixir) lib/enum.ex:116: Enumerable.reduce/3
       (elixir) lib/enum.ex:1627: Enum.reduce/3
       lib/phoenix_integration/requests.ex:612: anonymous fn/2 in PhoenixIntegration.Requests.merge_grouped_fields/2
       (stdlib) lists.erl:1263: :lists.foldl/3
       lib/phoenix_integration/requests.ex:612: anonymous fn/2 in PhoenixIntegration.Requests.merge_grouped_fields/2
       (stdlib) lists.erl:1263: :lists.foldl/3
       lib/phoenix_integration/requests.ex:290: PhoenixIntegration.Requests.submit_form/3
       lib/phoenix_integration/requests.ex:340: PhoenixIntegration.Requests.follow_form/3
       test/integrations/expense_integration_test.exs:15: (test)
```

**Comment by boydm** [comment]
I'd like to get a test in that covers the case. Can you please show the sample html input that is generating the error. I'll write the test, just show me what type of html input line you are using

**Comment by boydm** [comment]
I got it. I like the solution. I'll approve and add the test.

**Comment by martinos** [comment]
Here is an example:

``` html
<form accept-charset="UTF-8" action="/expenses" enctype="multipart/form-data" method="post">
  <input name="_csrf_token" type="hidden" value="BQEXVkMnOg1wfBxZBEI8JGFvMBp4EAAACuU3wvoLBLFko2Hc55aUJQ=="><input name="_utf8" type="hidden" value="✓">
  <div class="form-group receipt-view">
    <label for="expense_receipt">Receipt</label>
    <div class="form-file-button">
      <input class="button-file" id="expense_receipt" name="expense[receipt]" type="file">
    </div>
  </div>
  <div class="form-group">
    <button class="btn btn-primary" type="submit">Submit</button>
  </div>
</form>
```

I hope that it is what you need.

**Comment by boydm** [comment]
Thanks. I've rolled it all up and published it to hex as version 0.1.2
Could update your app and run it to let me know if all is good?
mix deps.update phoenix_integration
then rebuild and test

**Comment by martinos** [comment]
I have tested it with my app works like a charm. Thanks again for your great work.
